lib/inat_inferrer.py

```diff
@@ -208,13 +219,23 @@ def geo_model_predict(self, lat, lng, debug=False):
         if self.geo_elevation_model is None:
             return None
 
-        # lookup the H3 cell this lat lng occurs in
-        h3_cell = h3.geo_to_h3(float(lat), float(lng), 4)
-        h3_cell_centroid = h3.h3_to_geo(h3_cell)
-        # get the average elevation of the above H3 cell
-        elevation = self.geo_elevation_cells.loc[h3_cell].elevation
-        geo_scores = self.geo_elevation_model.predict(
-            h3_cell_centroid[0], h3_cell_centroid[1], float(elevation))
+        geo_scores = None
+        if encoder == "h3":
+            # lookup the H3 cell this lat lng occurs in
+            h3_cell = h3.geo_to_h3(float(lat), float(lng), 4)
+            h3_cell_centroid = h3.h3_to_geo(h3_cell)
+            # get the average elevation of the above H3 cell
+            elevation = self.geo_elevation_cells.loc[h3_cell].elevation
+            geo_scores = self.geo_elevation_model.predict(
+                h3_cell_centroid[0],
+                h3_cell_centroid[1],
+                float(elevation)
+            )
+        elif encoder == "raster" and self.coord_encoder is not None:
+            stacked_loc = np.array([[float(lng), float(lat)]])
+            encoded_loc = self.coord_encoder.encode(stacked_loc)
+            geo_scores = self.geo_elevation_model.predict_encoded(encoded_loc)
```

Comment:
At some point I think I'd like to refactor things so the TFGeoPriorModelElev class accepts a lat/lng and elevation encoding type and does the elevation calculation and scoring itself. That way TFGeoPriorModelElev can call CoordEncoder if it needs to, and this InatInferrer class doesn't need to think about those things. InatInferrer will just ask for the geo score for a lat lng using the traditional H3 encoder, or using the new CoordEncoder encoder and TFGeoPriorModelElev will handle the rest